### PR TITLE
Call handleEvent if it exists

### LINKED ISSUE #81: Event patcher doesn't work with EventTargets
If doing something like:

```js
class MyElement extends HTMLElement {
  connectedCallback() {
    this.shadowRoot.querySelector('.something').addEventListener('click', this);
  }

  handleEvent(ev) {

  }
}
```

`this` is the element. The method that should be called is actually `this.handleEvent`.

### PR DESCRIPTION
**Pull request by matthewp**
This fixes #81. If a handleEvent method exists, call that.

### CODE REVIEW

File: src/patch-events.js [hunk 1]
```diff
@@ -275,6 +275,10 @@ export function addEventListener(type, fn, optionsOrCapture) {
           return;
         }
       }
+      if(fn.handleEvent) {
```
Comment (azakus) [inline src/patch-events.js]: If a function has the `handleEvent` property added, only the function body is invoked by the event listener.

```js
let handler = typeof fn === 'function' ? fn : fn.handleEvent;
if (handler) {
  return handler.call(this, e);
}
```

Even better, the type of the handler should be determined in `addEventListener`

Comment (matthewp) [inline src/patch-events.js]: The `this` value in an EventTarget is the object containing the handleEvent function, not the element from which the event derives.

Comment (azakus) [inline src/patch-events.js]: Hmm, this seems to work counter to my reading of the specification: https://dom.spec.whatwg.org/#concept-event-listener-inner-invoke.

But this JSBin example shows that you are correct: http://jsbin.com/gagoyof/edit?html,console

Regardless, my first comment still stands. A function with `handleEvent` property still executes the function body: http://jsbin.com/jodolar/edit?html,console

Comment (matthewp) [inline src/patch-events.js]: So you want to change the condition so that it checks that `typeof fn !== 'function'`?

Comment (sorvell) [inline src/patch-events.js]: ^^ Yeah, I think that makes sense. That and a test would be great. Thanks.

Comment (matthewp) [inline src/patch-events.js]: The tests are currently failing on master: https://github.com/webcomponents/shadydom/issues/80

**Comment by sorvell** [comment]
Thanks for the PR and sorry for the long delay (will try to be more responsive moving forward!). Went ahead and incorporated this into https://github.com/webcomponents/shadydom/pull/131 and added tests.

**Pull request closed without merging**